src/commands/omnistudio/migration/migrate.ts

```diff
@@ -67,7 +76,33 @@ export default class Migrate extends OmniStudioBaseCommand {
 
     // Let's time every step
     DebugTimer.getInstance().start();
+    // const includeApex = this.flags.apex
+    //   ? await this.ux.confirm('Do you want to include Apex migration? (yes/no)')
+    //   : false;
 
+    if (relatedObjects) {
+      const validOptions = ['apex', 'lwc'];
+      const objectsToProcess = relatedObjects.split(',').map((obj) => obj.trim());
+      // Validate input
+      for (const obj of objectsToProcess) {
+        if (!validOptions.includes(obj)) {
+          throw new Error(`Invalid option provided for -r: ${obj}. Valid options are apex, lwc.`);
+        }
+      }
+      // Ask for user consent
+      const consent = await this.ux.confirm(
+        'By proceeding further, you hereby consent to the use, accept changes to your custom code, and the accompanying terms and conditions associated with the use of the OmniStudio Migration Tool. Do you want to proceed?'
+      );
+      if (!consent) {
+        this.ux.log('User declined consent. Aborting the process.');
+        return;
```

Comment:
Instead of stop processing complete migration, we should skip processing corresponding part ( Apex / LWC)